frontend/packages/shared/src/navigation/PackagesRouter/PackagesRouter.ts

```diff
@@ -0,0 +1,47 @@
+type ParamsOptions = {
+  org?: string;
+  app?: string;
+};
+
+type PackagesRoute =
+  | 'dashboard'
+  | 'editorOverview'
+  | 'editorUiEditor'
+  | 'preview'
+  | 'editorPublish';
+
+const packagesRoutes: Record<PackagesRoute, string> = {
+  dashboard: '/dashboard',
+  editorOverview: '/editor/{{org}}/{{app}}/overview',
+  editorUiEditor: '/editor/{{org}}/{{app}}/ui-editor',
+  editorPublish: '/editor/{{org}}/{{app}}/deploy',
+  preview: '/preview/{{org}}/{{app}}',
+};
+
+export class PackagesRouter {
+  private app: string;
+  private org: string;
+
+  constructor(private paramsOptions: ParamsOptions) {
+    this.app = this.paramsOptions.app ?? '';
+    this.org = this.paramsOptions.org ?? '';
+  }
```

Comment:
What about making the paramsOptions optinal to avoid sending empty objects to the constructor? 😊 

```suggestion
  constructor(private paramsOptions?: ParamsOptions) {
    this.app = this.paramsOptions.app ?? '';
    this.org = this.paramsOptions.org ?? '';
  }
```